### ISSUE TITLE
Full, running, multi-strategy authentication example, both client and server?

### ISSUE THREAD

**Issue by michaeladamkatz** [feathersjs/feathers#1167]
This is a very high level question. Please remove if it's not appropriate for a github "issue".

I have looked at many tutorials and examples of feathers authentication (in my case for a web application, not a native mobile app), but I haven't yet been able to find or create a full, working example. I feel like I'm looking for the most common use case, which is to allow users to be able to create an account and sign in to application with as little friction as possible, especially via an existing facebook account, google account, twitter account, but also with an email/password, including a confirmation email. I have read about feathers supporting all of these things, but some of it seems to be supported only through various extensions. All of the examples and tutorials I've seen (whether on Medium, the feathers documentation itself, "feathers awesome" and other github extension projects) are incomplete in some way, and it feels like a research project to tie it all together in the proper way, especially considering that many of the tutorials and partial examples seem to get outdated as best practices change and things that were extensions become integrated into the main library, etc.

I do understand that feathers is more do-it-yourself and open-ended than, say, firebase (e.g., in that the with feathers the coder has to provide a database), but it seems to me like the common use case of authentication is common enough that there should be a complete working example with all the standard workflows and caveats to look at. (From the display of the initial "sign up / login" screen, through the various workflow transition screens for the main OAuth providers, to signing out, to email confirmation, to lost passwords, and so on.) When I look at the documentation for implementing firebase authentication (https://firebase.google.com/docs/auth/web/firebaseui) I see exactly how to do all of it. Is there already a demo / reference implementation to see feathers authentication working in action? My preference would be for this example to have as few dependencies on other client or server frameworks as possible, and use a very standard database like postgres or mondo. Can there be (or is there already) something like "feathersjsdemo.com" that demonstrates all the feathers goodness in action? Like it could be a an app where users register, show off what they've built in feathers, participate in chat and Q/A, etc.

**Comment by MohammedFaragallah** [comment]
@michaeladamkatz any luck with this i'm trying to achieve the same thing.

**Comment by michaeladamkatz** [comment]
@MohammedFaragallah Nope, no response. I have decided to use CloudFlare for this purpose instead.

**Comment by daffl** [comment]
There is a GitHub example for the upcoming v4 in https://github.com/feathersjs/feathers-chat/pull/111. Adding multiple is as easy as adding the provider in the `oauth` configuration section. Accounts can now also be [linked to multiple providers](https://crow.docs.feathersjs.com/api/authentication/oauth.html#account-linking)

**Comment by MohammedFaragallah** [comment]
Thank you @daffl i'm really excited for v4
